app/middlewares/checkAuthentication.js

```diff
@@ -7,36 +7,35 @@ const logger = require('../logger');
 
 const SECRET_KEY = config.common.api.secretKey;
 
-const decode = token => {
+const decodeToken = async token => {
   try {
-    const result = jwt.decode(token, SECRET_KEY);
-    return models.user
-      .findOne({
-        where: { email: result.email }
-      })
-      .then(user => {
-        if (!user) {
-          throw errors.invalidToken();
-        }
+    const result = await jwt.decode(token, SECRET_KEY);
+    return result;
+  } catch (error) {
+    throw errors.invalidToken();
+  }
+};
 
-        return models.session.findOne({ where: { userId: user.id } }).then(session => {
-          if (!session || session.id !== result.token) {
-            throw errors.invalidToken();
-          }
-          return user;
-        });
-      })
-      .catch(() => {
-        throw errors.invalidToken();
-      });
-  } catch (e) {
+const decode = async decodeResult => {
```

Comment:
I like the changes, but this is no longer a decode function. It validates.

Reply:
ajajja i forgot to change it :P